overlord/ifacestate/ifacemgr.go

```diff
@@ -156,15 +164,21 @@ func (m *InterfaceManager) Ensure() error {
 	return nil
 }
 
-// Stop implements StateWaiterStopper.Stop. It stops
-// the udev monitor, if running.
+// Stop implements StateStopper. It stops the udev monitor,
+// if running.
 func (m *InterfaceManager) Stop() {
-	if m.udevMon == nil {
+	m.udevMonMu.Lock()
+	udevMon := m.udevMon
+	m.udevMonMu.Unlock()
+	if udevMon == nil {
 		return
 	}
-	if err := m.udevMon.Stop(); err != nil {
+	if err := udevMon.Stop(); err != nil {
 		logger.Noticef("Cannot stop udev monitor: %s", err)
 	}
+	m.udevMonMu.Lock()
+	defer m.udevMonMu.Unlock()
+	m.udevMon = nil
```

Comment:
I don't know if this is a realistic problem but in the case that Stop is concurrently executing with something that assigns to `m.udevMon` while the lock is not held, then the section here, after the lock is re-acquired might behave incorrectly. Perhaps it should check that `udevMon` is still stored in `m.udevMon` and only assign `nil` if that is the case?

Reply:
we don't run Ensure (the place where we set udevMon) while we are trying to Stop managers.